# --- SDE-COPYRIGHT-NOTE-BEGIN ---
# This copyright note is auto-generated by ./scripts/Create-CopyPatch.
# 
# Filename: package/.../hping/bytesex.h.patch
# Copyright (C) 2006 The OpenSDE Project
# 
# More information can be found in the files COPYING and README.
# 
# This patch file is dual-licensed. It is available under the license the
# patched project is licensed under, as long as it is an OpenSource license
# as defined at http://www.opensource.org/ (e.g. BSD, X11) or under the terms
# of the GNU General Public License as published by the Free Software
# Foundation; either version 2 of the License, or (at your option) any later
# version.
# --- SDE-COPYRIGHT-NOTE-END ---

This patch fixes byte ordering for amd64 and hppa.

it was taken from the gentoo linux repository:
http://sources.gentoo.org/viewcvs.py/gentoo-x86/net-analyzer/hping/files/ 

diff -uNr hping2-rc3.orig/bytesex.h hping2-rc3/bytesex.h
--- hping2-rc3.orig/bytesex.h	2004-07-29 15:46:47.572596272 +0000
+++ hping2-rc3/bytesex.h	2004-07-29 15:48:14.557372576 +0000
@@ -9,11 +9,14 @@
 
 #if 	defined(__i386__) \
 	|| defined(__alpha__) \
+	|| defined(__x86_64__) \
+	|| defined(__ia64__) \
 	|| (defined(__mips__) && (defined(MIPSEL) || defined (__MIPSEL__)))
 #define BYTE_ORDER_LITTLE_ENDIAN
 #elif 	defined(__mc68000__) \
 	|| defined (__sparc__) \
 	|| defined (__sparc) \
+	|| defined (__hppa__) \
 	|| defined (__PPC__) \
 	|| defined (__BIG_ENDIAN__) \
 	|| (defined(__mips__) && (defined(MIPSEB) || defined (__MIPSEB__)))
